ubuntu_image/parser.py

```diff
@@ -301,41 +313,59 @@ def parse(stream_or_string):
             offset_write = structure.get('offset-write')
             size = structure['size']
             structure_type = structure['type']
-            # Validate structure types.  These can be either GUIDs, two hex
-            # digits, hybrids, or the special 'mbr' type.  The basic syntactic
-            # validation happens above in the Voluptuous schema, but here we
-            # need to ensure cross-attribute constraints.  Specifically,
-            # hybrids and 'mbr' are allowed for either schema, but GUID-only
-            # is only allowed for GPT, while 2-digit-only is only allowed for
-            # MBR.  Note too that 2-item tuples are also already ensured.
+            structure_role = structure.get('role')
+            # Validate structure types and roles. These can be either GUIDs,
+            # two hex digits or hybrids.  The basic syntactic validation
+            # happens above in the Voluptuous schema, but here we need to
+            # ensure cross-attribute constraints.  Specifically, hybrids are
+            # allowed for either schema, but GUID-only is only allowed for GPT,
+            # while 2-digit-only is only allowed for MBR roles.
+            # Note too that 2-item tuples are also already ensured.
+            # We also still support the deprecated special type 'mbr' for
+            # legacy purposes, but issue a warning
+            if structure_type == 'mbr':
+                warn('Structure type mbr is deprecated, please use structure '
+                     'roles instead', DeprecationWarning)
```

Comment:
What about using the same nomenclature we use for GadgetSpecificationErrors?  E.g.

```
warn('volumes:<volume name>:structure:<N>:type = 'mbr' is deprecated; use role instead')
```

Or does that look too ugly?